packages/gguf/src/gguf.spec.ts

```diff
@@ -266,4 +273,12 @@ describe("gguf", () => {
 		const { parameterCount } = await ggufAllShards(URL_SHARDED_GROK);
 		expect(parameterCount).toEqual(316_490_127_360); // 316B
 	});
+
+	it("parse quant label", async () => {
+		expect(parseGGUFQuantLabel("Meta-Llama-3.1-8B-Instruct-Q4_K_M.gguf")).toEqual("Q4_K_M");
+		expect(parseGGUFQuantLabel("subdir/Meta-Llama-3.1-8B-Instruct-Q4_K_M.gguf")).toEqual("Q4_K_M");
+		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-Q2_K.gguf")).toEqual("Q2_K");
+		expect(parseGGUFQuantLabel("Codestral-22B-v0.1gguf")).toEqual(undefined);
+		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-F32-Q2_K.gguf")).toEqual("Q2_K"); // gguf name with two quant labels [F32, Q2_K]
```

Comment:
Some interesting test cases that can be added:

```suggestion
		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-F32-Q2_K.gguf")).toEqual("Q2_K"); // gguf name with two quant labels [F32, Q2_K]
		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-F32-IQ3_XS.gguf")).toEqual("IQ3_XS");
		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-F32-Q4_0_4_4.gguf")).toEqual("Q4_0_4_4");
```

Reply:
I had to push https://github.com/huggingface/huggingface.js/pull/967/commits/ed2a0df0ce964f1c75952aefaa0166d2c4593815

because `parseGGUFQuantLabel` is based on what's typed in [enum ggml_type](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md?plain=1#L133-L164). And `IQ3_XS `, `Q4_0_4_4` does not exist in [enum ggml_type](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md?plain=1#L133-L164) atm. However, I do see that [Codestral-22B-v0.1-IQ3_XS.gguf](https://huggingface.co/bartowski/Codestral-22B-v0.1-GGUF/blob/main/Codestral-22B-v0.1-IQ3_XS.gguf) is on the hub. I'd say: we should merge this PR and try to handle these edge cases in subseq PR